gen_errata.rb

```diff
@@ -369,11 +369,48 @@ def gen_ubuntu_errata(usn_db, packages, release_whitelist=nil, architecture_whit
         usn['releases'].each do |rel, dat|
           next if release_whitelist.is_a?(Array) && !release_whitelist.include?(rel)
 
-          unless dat.key?('archs')
-            warn "#{name} has no architectures for release #{rel}" if @verbose
-            next
+          # TODO: ESM-Updates do not have 'archs' -> Fallback to alternative handling
+          if dat.key?('archs')
+            add_packages_ubuntu(erratum, rel, dat, architecture_whitelist, packages)
+          elsif dat.key?('sources')
+            # try to do it the debian-way: get_binary_packages_for_erratum_package()
+            warn "#{name} has no architectures for release #{rel} -> using source-packages as fallback" if @verbose
+            dat['sources'].each do |source_pkg, pkg|
+              package = {}
+              pkg.each do |k, v|
+                package[k.to_sym || key] = v
+              end
+              package[:release] = rel
+
+              # skip if source_pkg not in packages_by_name
+              unless packages_by_name.key? source_pkg
+                warn "#{source_pkg} not in package-list; for #{name} and release #{rel} -> skipping" if @verbose
+                next
+              end
+
+              # set source-package
+              # FIXME: if multiple sources are specified by USN, we currently only take the last one!
```

Comment:
I guess, this shouldn't be an issue because one package should only be built from one source pkg.